tools/pd-heartbeat-bench/main.go

```diff
@@ -187,9 +187,21 @@ func createHeartbeatStream(ctx context.Context, cfg *config.Config) (pdpb.PDClie
 	}
 
 	go func() {
-		// do nothing
+		failedRequest := 0
 		for {
-			stream.Recv()
+			resp, err := stream.Recv()
+			if err != nil {
+				log.Error("receive error", zap.Error(err))
+				failedRequest++
+			}
+			if resp.GetHeader().GetError() != nil {
+				log.Error("receive error", zap.String("err", resp.GetHeader().GetError().String()))
+				failedRequest++
+			}
+			if failedRequest > 100 {
+				log.Error("receive error", zap.String("err", "too many errors"))
+				break
```

Comment:
I think no need to break.

Reply:
updated https://github.com/tikv/pd/pull/9224/commits/24111bdda740e934ef0ff99d05436df5ac2bf506